Dockerfile

```diff
@@ -0,0 +1,35 @@
+FROM debian:trixie
+
+RUN apt-get update && apt-get upgrade -y
+
+RUN apt-get update && apt-get install -y sudo openssh-server
+
+RUN apt-get update && apt-get install -y curl less vim
+
+# git is needed for parsl to figure out it's own repo-specific
+# version string
+RUN apt-get update && apt-get install -y git
+
+# useful stuff to have around
+RUN apt-get update && apt-get install -y procps
+
+# for building documentation
+RUN apt-get update && apt-get install -y pandoc
+
+# for monitoring visualization
+RUN apt-get update && apt-get install -y graphviz wget
+
+# for commandline access to monitoring database
+RUN apt-get update && apt-get install -y sqlite3
+
+RUN apt-get update && apt-get install -y python3.12 python3.12-dev
+RUN apt-get update && apt-get install -y python3.12-venv
```

Comment:
this is a bit weird (something to do with how python is packaged in trixie?). Normally in debian it looks like there's a single OS-level `python3` available (which changes when there's a new code-named release) and it seems unusual that trixie happens to have two. certainly debian isn't traditionally set up to expect you to be able to choose a python version from the OS.

there's a couple of things that could happen: i) always use the OS-level default `python3` or ii) use something like Conda to provide a much richer Python environment. Some dependencies like the ndcctools recommends being installed using conda anyway, and so maybe that's the way to go here. I don't think there's any particular reason to want to stick with the OS-level Python, as this is "an image where Parsl works" rather than "an image that looks like a particular debian version".